Add `needs:` syntax example using `always()` condition (#1307)
github/docs: content/actions/reference/workflow-syntax-for-github-actions.md

2. `job2`
3. `job3`

> #### Example not requiring dependent jobs to be successful

Reviewer: I'll raise a docs issue for all the titles in this guide, so that we can move them to an "Example: `verb`" convention

Reviewer: Done: https://github.com/github/docs/issues/2063 👍

```yaml
jobs:
  job1:
  job2:
    needs: job1
  job3:
    if: always()
    needs: [job1, job2]
```

> In this example, `job3` uses the `always()` conditional expression so it always runs after `job1` and `job2` completes, whether `job1` and `job2` are successful or not. For more information, see "[Context and expression syntax](/actions/reference/context-and-expression-syntax-for-github-actions#job-status-check-functions)."

Reviewer: ```suggestion
In this example, `job3` uses the `always()` conditional expression so it that it always runs after `job1` and `job2` have completed, regardless of whether they were successful. For more information, see "[Context and expression syntax](/actions/reference/context-and-expression-syntax-for-github-actions#job-status-check-functions)."
```

Reviewer 2: Sure! Tiny typo fix on your suggestion:

```suggestion
In this example, `job3` uses the `always()` conditional expression so that it always runs after `job1` and `job2` have completed, regardless of whether they were successful. For more information, see "[Context and expression syntax](/actions/reference/context-and-expression-syntax-for-github-actions#job-status-check-functions)."
```
